test/langtools/jdk/javadoc/doclet/testDocletExample/TestDocletExample.java

```diff
@@ -89,6 +94,9 @@ public void testDocletExample(Path base) throws Exception {
         var docletPkg = snippets.getElements().getPackageElement("jdk.javadoc.doclet");
         var dc = snippets.getDocTrees().getDocCommentTree(docletPkg);
         var exampleSnippet = snippets.getSnippetById(dc, "Example.java");
+        if (exampleSnippet == null) {
```

Comment:
Same as above.

Reply:
Yes, same reasoning. `SnippetUtils` is broken if we get a null result.

Reply:
> `SnippetUtils` is broken if we get a null result.

If it's unconditionally so, which it seems to be, wouldn't it be better to throw something from `SnippetUtils.getSnippetById` instead?

Reply:
Hmmm, The current spec of `SnippetUtils` is to return `null` if an item cannot be found.  So, it is only broken if the user really expects a non-null result.

I guess it depends if there is enough use case where you might want a null result instead of an exception.

Reply:
I guess the `SnippetUtils` spec does not currently define behavior if not found.

I would be more inclined to define a "returns null" behavior than to use `Optional` or throw and exception.   But more generally, I'd prefer to leave updates to `SnippetIUtils` to its own separate Enhancement.